okhttp/src/main/java/okhttp3/internal/http/RetryAndFollowUpInterceptor.java

```diff
@@ -164,13 +169,14 @@ public StreamAllocation streamAllocation() {
       }
 
       if (followUp.body() instanceof UnrepeatableRequestBody) {
+        streamAllocation.release();
```

Comment:
I think this is the right move. As I mentioned, a few lines up the response body gets closed. That boils down to a call to StreamAllocation#streamFinished which nulls out the codec. It will not release the connection because StreamAllocation#release has not yet been called. So this call to release will return the connection to the pool. Let me know if you think that's inaccurate.